docs/DeveloperGuide.md

```diff
@@ -7,32 +7,97 @@
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Basic Class Diagram 
+![image](https://user-images.githubusercontent.com/53790951/160271319-4a351f51-afd7-4e04-9451-f04143146551.png)
```

Comment:
I think more descriptive roles can be used to describe these relationships, since things like `uses` or `instantiates` can already be implied by the arrows? Also, if `instantiates` implies an association, consider using a solid line for it instead.

![image](https://user-images.githubusercontent.com/8340533/160965280-da7e23ba-e500-4c69-bf67-a89075e2e903.png)